webboot/webboot_test.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"io/ioutil"
+	"os"
+	//	"log"
+	"path/filepath"
+	"strings"
+	"testing"
+	//	"reflect"
+)
+
+type test struct {
+	name         string
+	linkOrName   string
+	md5link      string
+	expectedLink string
+	expectedName string
+}
+
+func TestParseArgs(t *testing.T) {
+	tests := []test{
+		{name: "Debian", linkOrName: "https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-10.0.0-amd64-netinst.iso", expectedLink: "https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-10.0.0-amd64-netinst.iso", expectedName: "debian-10.0.0-amd64-netinst.iso"},
+		{name: "TinyCore", linkOrName: "tinycore", expectedLink: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", expectedName: "tinycore"},
+		{name: "Ubuntu", linkOrName: "ubuntu", expectedLink: "http://releases.ubuntu.com/18.04.2/ubuntu-18.04.2-desktop-amd64.iso", expectedName: "ubuntu"},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			link, filename, err := parseArgs(v.linkOrName)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if strings.Compare(link, v.expectedLink) != 0 {
+				t.Errorf("Failed to receive the correct link: got %v, want %v", link, v.expectedLink)
+			}
+			if strings.Compare(filename, v.expectedName) != 0 {
+				t.Errorf("Failed to receive the correct file name: got %v, want %v", filename, v.expectedName)
+			}
+		})
+	}
+}
+
+func TestName(t *testing.T) {
+	tests := []test{
+		{name: "TinyCore", linkOrName: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", expectedName: "CorePure64-10.1.iso"},
+		{name: "ArchLinux", linkOrName: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/archlinux-2019.08.01-x86_64.iso", expectedName: "archlinux-2019.08.01-x86_64.iso"},
+		{name: "Ubuntu", linkOrName: "http://releases.ubuntu.com/18.04.2/ubuntu-18.04.2-desktop-amd64.iso", expectedName: "ubuntu-18.04.2-desktop-amd64.iso"},
+		{name: "HtmlFile", linkOrName: "http://releases.ubuntu.com", expectedName: "index.html"},
+	}
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			filename, err := name(v.linkOrName)
+			if err != nil {
+				t.Fatalf("Invalid URL: got %v, want a valid URL", v.linkOrName)
```

Comment:
Oh, I see what you mean. I changed it. Thank you

Reply:
I changed it to that it to follow this format

Reply:
I see what you mean by what I should be printing out as an error message since the testing is done for err != nil. This message could be worded better for testing